packages/editor-ui/src/components/Node.vue

```diff
@@ -261,12 +292,21 @@ export default defineComponent({
 		const deviceSupport = useDeviceSupport();
 		const { callDebounced } = useDebounce();
 
+		const nodeBase = useNodeBase({
+			name: props.name,
+			instance: props.instance,
+			workflowObject: props.workflow,
+			isReadOnly: props.isReadOnly,
+			emit: emit as (event: string, ...args: unknown[]) => void,
+		});
+
 		return {
 			contextMenu,
 			externalHooks,
 			nodeHelpers,
 			pinnedData,
 			deviceSupport,
+			...nodeBase,
```

Comment:
nit: Would prefer without the spread, but I understand if it's to limit the amount of refactoring in this PR

Reply:
Good point. Fixed.

Reply:
Had to undo the change. Refs and typechecks don't play well with options API. 😰